src/toil/fileStores/__init__.py

```diff
@@ -14,6 +14,8 @@
 import os
 import stat
 
+from uuid import uuid4
+
 
 def make_public_dir(dirName: str) -> str:
```

Comment:
Is there a good reason we're making world-writable temp directories for Toil jobs? It looks like this function is used in the file stores as well. Are we expecting some jobs to run under another user?

If the directories do need to be world-writable, we probably need to set the sticky bit like on a real /tmp; I think otherwise anyone can use your write permission on the directory to delete anyone else's files.

Reply:
Arjun decided these directories should be public when first implementing caching: https://github.com/DataBiosphere/toil/commit/3974681ac5b903c967ca31b59d6890e20761859d#diff-bdafbc89001da41469f9aa17902123fa92515e69dfe87beb2643e48dbde8f2f0R583

I didn't see any justification there.

Reply:
You have some good points.  This is a discussion outside of the context of this PR, however, as this PR is just shortening names, not changing current behavior.

That said, just for discussion, an argument against changing the permissions of the directory that comes to mind to me would be inspection by an outside user on the dir/files of a failed workflow (mentioned above).  I also think that... while it's possible for an outside program to attempt to write to our toil created directory while it's running... in practice, I don't see it happening.

Reply:
This PR may be relevant to the historical discussion: https://github.com/DataBiosphere/toil/pull/3229